### DISCUSSION TITLE
Compiled dbt Preview button not working

### DISCUSSION THREAD

**Discussion by letuio** [AltimateAI/vscode-dbt-power-user#1225] [Q&A]
Hi everyone, 

I was looking for some help, since this option was working fine for me until yesterday afternoon. It was fine in the morning and for the life of me I can't figure out why it stopped in the middle of the day, as far as I know there shouldn't have been any changes.

The problem is the following, the button to see the compiled dbt preview stopped working and it is giving me the following issue, no matter the model that I am using. For example:
![image](https://github.com/AltimateAI/vscode-dbt-power-user/assets/152965967/a0232c27-f82b-4a5e-b263-b701c88c38cf)

It always seems to be the same problem, no matter the model. I am not sure what changed, but I've been using it for quite a while with no issues. I am able to run models and compile models using the right click menu or through the CLI, I can see the compiled code on the target folder. But this specific button is not working.

Seems to indicate that it expects a 'NodeInfo' object to have the 'keys' attribute, but I don't know anything about that. 

Any help would be appreciated, as I like this feature a lot and I use it constantly. Only work-around I've seen is to compile the code manually and then use the command to "open dbt target folder compiled" but that is hardly as convenient.

Thanks in advance.

**Comment by ngoc-hien** [comment]
Hi,
I now get error "'NodeInfo' object has no attribute 'keys'" when executing dbt SQL. It did not happen before.

**Reply by letuio** [reply]
ok so it's not just me at least. My teammates seem to not have the problem, though they don't use this feature much. I wonder what is causing it.

**Reply by ngoc-hien** [reply]
I installed the previous version and no error.

**Reply by letuio** [reply]
which version number was it? I assume you mean for dbt power user. I also tried going down one version from v0.40.1 to v0.40.0 but it didn't work. Did you go further back to version 0.39?

**Reply by ngoc-hien** [reply]
Yes, v0.39 for dbt power user.

**Reply by letuio** [reply]
Awesome, thanks I will try that. Maybe it's some new change that introduced this error. I am also on dbt core around 1.4, so this might not be an issue if you're around 1.8

**Comment by anandgupta42** [comment]
@ngoc-hien @letuio We are looking at the issue : https://github.com/AltimateAI/vscode-dbt-power-user/issues/1226

**Comment by ngoc-hien** [comment]
Hi,
I have upgraded dbt-clickhouse from version 1.4.1 to 1.7.1. Then updated dbt power user to v.0.40.3. Everything works well now. No issue when executing dbt SQL model. Maybe the issue is related to the old version of dbt-clickhouse.

**Reply by letuio** [reply]
Thanks for the heads up! I don't think I have dbt-clickhouse. It's actually the first time I hear about it, so maybe it is related but likely not the only reason?
